app/controllers/users_controller.rb

```diff
@@ -52,6 +52,20 @@ def redirect
 
   private
 
+  def set_cookie_and_redirect(email)
+    cookies[:h_email] = { value: email }
+    redirect_to '/refer-a-friend'
+  end
+
+  def user_exists
+    email = params[:user][:email].downcase
+    params[:user][:email] = email
+    user = User.find_by_email(email)
+    if not user.nil?
```

Comment:
also, while i'm not the maintainer, i'm sure it'd be appreciated if there was a unit test or two for this :)